### PR TITLE
OpcodeDispatcher: Handle VMOVAPS

### PR DESCRIPTION
**Pull request by lioncash**
Starts the AVX instruction implementation work. Also pls let me know if the decoder changes are A-OK. There's so much to keep track of mentally, I'm not sure if I missed anything.

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
I think the frontend changes are fine here. It would have been how I would have wrote it.
In the future we will need ASM tests that prepend VEX.L on instructions that don't support it to ensure testing, but seems fine for now.
Just need CI to go green.

**Comment by lioncash** [comment]
Looking at the x86 failure. Though the ARM builder failures have me stumped, considering the AVX bits shouldn't even be running on them aside from the simulator builders

**Comment by Sonicadvance1** [comment]
I think the VEX table needs to be moved to only be filled out when the host supports AVX, because we are hitting a host side SIGILL rather than a guest side SIGILL. That should resolve the ARM runners.

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp [hunk 1]
```diff
@@ -5724,7 +5724,7 @@ void OpDispatchBuilder::InstallHostSpecificOpcodeHandlers() {
   Initialized = true;
 }
 
-void InstallOpcodeHandlers(Context::OperatingMode Mode) {
+void InstallOpcodeHandlers(Context::OperatingMode Mode, Context::Context *CTX) {
```
Comment (Sonicadvance1) [inline External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:5727]: You don't want to do this in `InstallOpcodeHandlers`, you'll want to do this in `InstallHostSpecificOpcodeHandlers` just above this function.

Comment (lioncash) [reply External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:5727]: Oh, duh, missed that.

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp [hunk 2]
```diff
@@ -38,6 +38,18 @@ void OpDispatchBuilder::MOVAPSOp(OpcodeArgs) {
   StoreResult(FPRClass, Op, Src, -1);
 }
 
+void OpDispatchBuilder::VMOVAPSOp(OpcodeArgs) {
+  OrderedNode *Src = LoadSource(FPRClass, Op, Op->Src[0], Op->Flags, -1);
+  const auto Is128BitDest = GetDstSize(Op) == Core::CPUState::XMM_SSE_REG_SIZE;
+
+  if (Op->Dest.IsGPR() && Is128BitDest) {
+    // Perform 32-bit store to clear the upper lane.
```
Comment (Sonicadvance1) [inline External/FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp:46]: 32-bit or 32-byte?

Comment (lioncash) [reply External/FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp:46]: oh whoops, definitely meant byte there. Was probably thinking about 256-bit and 32-byte at the same time and both won out in their own way.

**Review by Sonicadvance1 — APPROVED** [review]
Nice nice nice nice nice.

**Pull request merged**